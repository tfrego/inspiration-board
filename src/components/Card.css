.card {
  background-color: #ffc;

  padding: 0 0;
  margin: 0.5rem;

  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  display: grid;
  grid-template-columns: 1fr 5fr 1fr;
  align-items: center;

  /* Firefox */
  -moz-box-shadow:5px 5px 7px rgba(33,33,33,1);
  /* Safari+Chrome */
  -webkit-box-shadow: 5px 5px 7px rgba(33,33,33,.7);
  /* Opera */
  box-shadow: 5px 5px 7px rgba(33,33,33,.7);

  /* tilt */
  -webkit-transform:rotate(-6deg);
  -o-transform:rotate(-6deg);
  -moz-transform:rotate(-6deg);

  /* smooth transitions */
  -moz-transition:-moz-transform .15s linear;
  -o-transition:-o-transform .15s linear;
  -webkit-transition:-webkit-transform .15s linear;
}

/* tilt and color */
.card:nth-child(even) {
  -o-transform:rotate(4deg);
  -webkit-transform:rotate(4deg);
  -moz-transform:rotate(4deg);
  position:relative;
  top:5px;
  background:#cfc;
}

/* tilt and color */
.card:nth-child(3n) {
  -o-transform:rotate(-3deg);
  -webkit-transform:rotate(-3deg);
  -moz-transform:rotate(-3deg);
  position:relative;
  top:-5px;
  background:#ccf;
}

.card:nth-child(5n) {
  -o-transform:rotate(5deg);
  -webkit-transform:rotate(5deg);
  -moz-transform:rotate(5deg);
  position:relative;
  top:-10px;
}

/* zoom on hover and focus */
/* z-index to ensure that the enlarged sticky note covers the others */
.card:hover, .card:focus{
  -moz-box-shadow:10px 10px 7px rgba(0,0,0,.7);
  -webkit-box-shadow: 10px 10px 7px rgba(0,0,0,.7);
  box-shadow:10px 10px 7px rgba(0,0,0,.7);
  -webkit-transform: scale(1.25);
  -moz-transform: scale(1.25);
  -o-transform: scale(1.25);
  position:relative;
  z-index:5;
}

.card__content {
  grid-column-start: 2;

  display: flex;
  flex-direction: column;
  justify-content: space-around;

  text-align: center;
  font-weight: 100;
  overflow: hidden;
}

.card__content-text {
  text-overflow: ellipsis;
  overflow: hidden;
  font-family: 'Reenie Beanie', cursive;
  font-size: 1.75rem;
}

.card__content-emoji {
  font-size: 3rem;
}

.card__content-text, .card__content-emoji {
  padding: 0;
  margin: 0;
}

.card__delete {
  align-self: start;
  border: none;
  background: none;
  text-align: left;
  font-size: large;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
}

.card__delete:hover {
  color: red;
}
