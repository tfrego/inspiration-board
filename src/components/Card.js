import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  render() {
    return (
      <div className="card">
        {this.props.text}
        {this.props.emoji}
      </div>
    )
  }
}

Card.propTypes = {
  text: PropTypes.string,
  emoji: PropTypes.string,
};

export default Card;
