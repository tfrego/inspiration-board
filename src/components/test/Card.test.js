import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('board', () => {
  it('will match the Board Snapshot', () => {
    const wrapper = shallow( <Card text="hello" emoji="heart_eyes" />);

    expect(wrapper).toMatchSnapshot();
  })
})
