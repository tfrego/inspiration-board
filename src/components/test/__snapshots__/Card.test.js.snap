// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`board will match the Board Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    emoji="heart_eyes"
    text="hello"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="card__content-text"
        >
          hello
        </p>,
        <p
          className="card__content-emoji"
        >
          üòç
        </p>,
      ],
      "className": "card card__content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "hello",
          "className": "card__content-text",
        },
        "ref": null,
        "rendered": "hello",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "üòç",
          "className": "card__content-emoji",
        },
        "ref": null,
        "rendered": "üòç",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="card__content-text"
          >
            hello
          </p>,
          <p
            className="card__content-emoji"
          >
            üòç
          </p>,
        ],
        "className": "card card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "hello",
            "className": "card__content-text",
          },
          "ref": null,
          "rendered": "hello",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "üòç",
            "className": "card__content-emoji",
          },
          "ref": null,
          "rendered": "üòç",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
